docs/contributing.md

```diff
@@ -139,7 +139,10 @@ Documentation changes are also welcome!
 To verify changes locally:
 
 ```bash
-poetry run mkdocs serve -a localhost:8001
+python3 -m venv .venv
```

Comment:
would it not mess up `venv` for docs and `venv` for the rest of the repo? I consider moving everything related to docs to `docs/`, including (or only?) `mkdocs.yml`

Then we can create `.python-version` file that can be picked up by `pyenv` and [github action](https://github.com/actions/setup-python/blob/main/docs/advanced-usage.md#using-the-python-version-file-input) and have the `docs/venv` for better isolation